### PR TITLE
Update to newly renamed ipyvtklink

### PR DESCRIPTION
**Pull request by banesullivan**
Use the newly renamed `ipyvtklink`: https://pypi.org/project/ipyvtklink/

### CODE REVIEW

**Review by akaszynski — APPROVED** [review]
This "works" locally for me, but I'm running into an issue with a frozen window.  I'm debugging my instance, but if I can't figure out what's causing it I'll post an issue on the appropriate repo.

**Comment by akaszynski** [comment]
> This "works" locally for me, but I'm running into an issue with a frozen window. I'm debugging my instance, but if I can't figure out what's causing it I'll post an issue on the appropriate repo.

Turns out it's a local issue.  We can merge this and hopefully get a release soon.

File: pyvista/jupyter/__init__.py [hunk 1]
```diff
@@ -84,17 +84,26 @@ def set_jupyter_backend(backend):
...
         raise ImportError('Install IPython to display with pyvista in a notebook.')
 
+    if backend == 'ipyvtk_simple':
+        try:
+            import ipyvtklink
+        except ImportError:
+            raise ImportError('Please install `ipyvtklink`. `ipyvtk_simple` '
+                              'is deprecated.')
+        else:
+            backend = 'ipyvtklink'
```
Comment (banesullivan) [inline pyvista/jupyter/__init__.py:94]: @akaszynski, is this well handled in your opinion?

Should we raise a warning here if the user keeps using the `ipyvtk_simple` name?

File: docs/conf.py [hunk 2]
```diff
@@ -126,8 +126,9 @@
     "doc_module": "pyvista",
     "image_scrapers": ('pyvista', 'matplotlib'),
     'first_notebook_cell': ("%matplotlib inline\n"
-                            "from pyvista import set_plot_theme\n"
-                            "set_plot_theme('document')"),
+                            "from pyvista import set_jupyter_backend, set_plot_theme\n"
+                            "set_plot_theme('document')\n"
+                            "set_jupyter_backend('ipyvtklink')"),
```
Comment (banesullivan) [inline docs/conf.py:131]: FYI, is this okay?

Comment (akaszynski) [reply docs/conf.py:131]: Yes.  It's always confusing for a new user why it's not working with dynamic plots.

Comment (akaszynski) [reply docs/conf.py:131]: In fact, I'd like to set it as default for the entire module.  Can we make `ipyvtklink` a hard dependency?

Comment (akaszynski) [reply docs/conf.py:131]: > In fact, I'd like to set it as default for the entire module. Can we make ipyvtklink a hard dependency?

Might not work out if this requires ipython at some point in the install...  Maybe warn the user?

Comment (banesullivan) [reply docs/conf.py:131]: I'm hesitant to add it as a hard dependency for those that want to use PyVista outside of ipython and don't want to install that stack as well. I think soft is the way to go with a clear error message or warning when folks try to use it.

Let's set it as the default and if it isn't available and the user is in Jupyter, throw a warning and fallback to the static plot

File: pyvista/jupyter/__init__.py (hunk 1 above)
Comment (akaszynski) [reply pyvista/jupyter/__init__.py:94]: Error is fine.  Usage of `ipyvtk_simple` as a backend was never in our API, so it's not even a depreciation error right now.  We can leave this in, but it's not even necessary at this point.

File: docs/conf.py (hunk 2 above)
Comment (akaszynski) [reply docs/conf.py:131]: Sounds good.  I already have people asking (both internally and externally) for interactive notebook plots.  Giving them a warning would help them with the installation.

Comment (banesullivan) [reply docs/conf.py]: See the latest changes. Now the code will always fallback to showing a static output:

Screen Shot 2021-04-14 at 4 12 24 PM

Comment (akaszynski) [reply docs/conf.py]: Perfect.  This will make sense to new and existing users.

**Comment by banesullivan** [comment]
Ready to merge?

**Comment by akaszynski** [comment]
Yes.  Fire when ready.

**Pull request merged**